### PR TITLE
API Updates

### PR DESCRIPTION
**Pull request by ColdenCullen**
Pairs well with awslabs/aws-c-mqtt#34.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_crt/io.py [hunk 1]
```diff
@@ -30,3 +30,72 @@ def __init__(self, elg):
...
+TlsVersion = type('TlsVersion', (), dict(
+    SSLv3 = 0,
+    TLSV1 = 1,
+    TLSV1_1 = 2,
+    TLSV1_2 = 3,
+    TLSV1_3 = 4,
+    Default = 128,
+))
+
+class TlsContextOptions(object):
```
Comment (justinboswell) [inline aws_crt/io.py]: You may want to add new_default_client and new_default_server

File: setup.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 if compiler_type == 'msvc':
     pass
 else:
-    cflags += ['-O0', '-fsanitize=address']
+    cflags += ['-O0', '-Wextra', '-Werror', '-fsanitize=address']
```
Comment (justinboswell) [inline setup.py]: I'm gonna overwrite this with my changes, but we should remove -fsanitize=address from the defaults

File: setup.py [hunk 3]
```diff
@@ -28,7 +28,7 @@
     ],
     include_dirs = ['/usr/local/include', os.getenv('AWS_C_INSTALL') + '/include'],
     library_dirs = ['/usr/local/lib', os.getenv('AWS_C_INSTALL') + '/lib'],
-    libraries = ['aws-c-common', 'aws-c-io', 'aws-c-mqtt'],
+    libraries = ['aws-c-mqtt', 'aws-c-io', 'aws-c-common', 's2n', 'crypto'],
```
Comment (justinboswell) [inline setup.py]: I fixed this as well, don't need s2n or crypto on mac

File: aws_crt/io.py [hunk 4]
```diff
@@ -30,3 +30,72 @@ def __init__(self, elg):
...
+
+        assert isinstance(pkcs12_path, str)
+        assert isinstance(pkcs12_password, str)
+
+        opt = TlsContextOptions()
+        opt.pkcs12_path = pkcs12_path
+        opt.pkcs12_password = pkcs12_password
+        return opt
+
+class ClientTlsContext(object):
```
Comment (justinboswell) [inline aws_crt/io.py]: In Java, this is just TlsContext, any reason to differentiate type-wise? If so, I'll make the same changes to Java.

**Comment by graebm** [comment]
Can you pretty pls add enums for QOS

**Pull request merged**